### PR TITLE
Add version to test scripts

### PR DESCRIPTION
**Pull request by glennj**
6 items in here, each in a separate commit:

1. revert .meta/version back to canonical version where I've incremented the patch version
1. add version to each exercise test file
1. add a brief comment where bash-specific tests have been added.
1. other comments noted while reviewing test files
1. bump version for robot-simulator (canonical data only changed descriptions, no tests)
1. update bin/canonical_data_check to look for mismatches between meta file and test file

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/bash/blob/master/POLICIES.md)

### CODE REVIEW

**Comment by glennj** [comment]
FYI, command to add version to test files:
```bash
bash -c '
    cd ~/src/exercism/bash/exercises
    for slug in */; do
        slug=${slug%/}
        test=${slug//-/_}_test.sh


        (
            cd "$slug" && {
                if   [[ -f ./.meta/version ]]
                then ver=$(< ./.meta/version)
                else ver="<none>"
                fi
                ed "$test" <<END_ED
1a

# canonical-data version: $ver
.
wq
END_ED
            }
        )
    done
'
```

File: exercises/pig-latin/pig_latin_test.sh [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env bash
 
+# canonical-data version: 1.2.0
```
Comment (guygastineau) [inline exercises/pig-latin/pig_latin_test.sh]: I'm a little confused. I thought the reason to have these here was to document that our files have deviated from the canonical-data versions?

File: exercises/acronym/acronym_test.sh [hunk 2]
```diff
@@ -63,6 +65,10 @@
   [[ "$output" == "TRNT" ]]
 }
 
+# bash-specific test: Focus the student's attention on the effects of 
+# word splitting and filename expansion:
+# https://www.gnu.org/software/bash/manual/bash.html#Shell-Expansions
+
```
Comment (guygastineau) [inline exercises/acronym/acronym_test.sh]: Good convention

File: exercises/pig-latin/pig_latin_test.sh (hunk 1 above)
Comment (glennj) [reply exercises/pig-latin/pig_latin_test.sh]: Good point. Will update

File: exercises/acronym/acronym_test.sh [hunk 3]
```diff
@@ -1,5 +1,8 @@
 #!/usr/bin/env bash
 
+# canonical-data version: 1.7.0
+# plus additional track-specific tests
```
Comment (guygastineau) [inline exercises/acronym/acronym_test.sh]: So, if new tests are added I don't see this giving us information about some sort of local patch version.  Do you want a local patch version?  Or maybe there could be a local minor and patch version?

**Comment by glennj** [comment]
I think I accidentally resolved a conversation.

I see your point (finally) about local track version. I'll update all the test versions with a 4th dot-separated version number, zero where there's no update, one if there are track-specific tests.

**Comment by guygastineau** [comment]
Great work @glennj I will conform my other PR to these conventions.  Unless it is in here, and I just missed it while reviewing.

**Comment by glennj** [comment]
> Great work @glennj I will conform my other PR to these conventions. Unless it is in here, and I just missed it while reviewing.

It is here: https://github.com/exercism/bash/pull/399/files#diff-c0f3ee502053f8111604759048081d15

**Pull request merged**